### ISSUE TITLE
[22.03] FW4 - interfaces with firewall rules require manual firewall restart

### ISSUE THREAD

**Issue by hitech95** [openwrt/openwrt#9611]
Me and many other with IPv4aaS that are using the map package are facing an
issue with the new fw4.

After teh new interface is created a manual restart of the firewall service is required.
looks like that netifd does not reload the service.

Devices tested:
- X86 (VM)
- ARM64 (RPI4)

We have tested with the new branched 22.03. We are going to perform some more tests with the latest snapshot.
and if possible on other HW too!

**Comment by hitech95** [comment]
> In particular check if `fw4 -q network "$INTERFACE"` succeeds.

I'm absolutly new with fw4/nftables. Can you provide some example/reference code to better understand what you mean?

**Comment by edofullin** [comment]
This is what I get from logread upon network restart, it seems like the firewall reload only happens because the wan.101 interface changes state regardless of what map-wan6_4.

In this case everything works because the map interface was already created, but I wonder if the issue might be caused by the reload trigger from wan.101 happening before the map interface is setup.

Who should call the firewall reload? I do not see such calls in map.sh, which is the script responsible for setting up the map interface. 

```
Fri Apr  1 10:46:23 2022 daemon.notice netifd: Interface 'wan6_4' is setting up now
Fri Apr  1 10:46:23 2022 kern.info kernel: [ 1403.573828] nat46: adding device (map-wan6_4)
Fri Apr  1 10:46:23 2022 kern.debug kernel: [ 1403.578220] [nat46] make_nat46_instance: allocated nat46 instance with 1 pairs
Fri Apr  1 10:46:23 2022 kern.debug kernel: [ 1403.586058] nat46: netdevice nat46 'map-wan6_4' created successfully.
Fri Apr  1 10:46:23 2022 user.notice firewall: Reloading firewall due to ifup of wan6 (wan.101)
Fri Apr  1 10:46:23 2022 kern.info kernel: [ 1403.608765] nat46: configure device (map-wan6_4) with 'local.style MAP local.v4 101.58.48.0/20 local.v6 2a0e:410:3000::/36 local.ea-len 12 local.psid-offset 6 remote.v4 0.0.0.0/0 remote.v6 2a0e:402:101:100::/64 remote.style RFC6052 remote.ea-len 0 remote.psid-offset 0'
Fri Apr  1 10:46:23 2022 kern.info kernel: [ 1403.652075] nat46: insert new rule into device (map-wan6_4) with 'remote.style MAP remote.v4 101.58.48.0/20 remote.v6 2a0e:410:3000::/36 remote.ea-len 12 remote.psid-offset 6'
Fri Apr  1 10:46:23 2022 kern.debug kernel: [ 1403.667835] [nat46] make_nat46_instance: allocated nat46 instance with 2 pairs
Fri Apr  1 10:46:23 2022 kern.debug kernel: [ 1403.675105] [nat46] release_nat46_instance: freeing nat46 instance with 1 pairs
Fri Apr  1 10:46:23 2022 daemon.notice netifd: Interface 'wan6_4' is now up
Fri Apr  1 10:46:23 2022 daemon.notice netifd: Network device 'map-wan6_4' link is up
```

**Comment by jow-** [comment]
> Who should call the firewall reload?

The hotplug script  Imentioned above should call firewall reload upon any ifup event for a `$INTERFACE` that is covered by any zone of `/etc/config/firewall`. The `ifup` event is emitted by netifd.

**Comment by jow-** [comment]
Fixed with 2df17604a4f891447beb66988e3d83e23ab3a3b0
